src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1185,6 +1185,9 @@ else if (fieldClass == String[].class) {
                     if ("fileExtensions".equals(propertyName)) {
                         result = "all files";
                     }
+                    else if ("customImportOrderRules".equals(propertyName)) {
```

Comment:
I don't believe we are expected to create a special override for just this property.

@romani Please confirm.

Reply:
I do not expect this change.
@wltan , please explain.

Reply:
@romani This goes back to my post in https://github.com/checkstyle/checkstyle/issues/7128#issuecomment-536707069 .
xdoc display is strictly controlled by UT and this code is related to the default value if it sees an empty list.

Reply:
> xdoc display is strictly controlled by UT and this code is related to the default value if it sees an empty list.

Yes, the UT wouldn't let me change the value to `""` otherwise.

I am however open to changing where exactly we want to make this change. Note how the changes above are in the `else if (fieldClass == String[].class) {` block, even if our property is set using a single `String`. Refer to this snippet:
https://github.com/checkstyle/checkstyle/blob/aa5a9fd19a6c2e05dbfa756e259f7f59b7d676ce/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L857-L864
The root cause is that right now the field for `customImportOrderRules` is a `List<String>`, so `getFieldClass` is returning `String[].class`. If you want it to be `String.class` instead, then some additional modifications will have to be made.

Reply:
Check should reserve user defined value as is, so field should be changed to have String type `String customImportOrderRules;`.
But we need to create new field `List<String> customOrderRules;` that keeps parsed value from user input, user defined value will not be used in Check at all , except for creation of customOrderRules in setter.

Reply:
I've made the requested changes and the hardcoding above is no longer required, so it has been removed.

There were some minor problems with PMD which doesn't like the field to be unused so I made a few adjustments to `setCustomImportOrderRules` to use and assign to it.